src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,28 @@
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        lock.readLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+            lock.readLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        return list.get(index);
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        int size;
+        lock.writeLock().lock();
+        try {
+            size = list.size();
+        } finally {
+            lock.writeLock().unlock();
```

Comment:
Unlock the read lock instead of the write lock for the `size` method. Replace `lock.writeLock().unlock();` with `lock.readLock().unlock();`. 🌟